lib/msal-common/src/client/RefreshTokenClient.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Configuration } from "../config/Configuration";
+import {BaseClient} from "./BaseClient";
+import {RefreshTokenRequest} from "../request/RefreshTokenRequest";
+import {Authority, NetworkResponse} from "..";
+import {ServerAuthorizationTokenResponse} from "../server/ServerAuthorizationTokenResponse";
+import {RequestParameterBuilder} from "../server/RequestParameterBuilder";
+import {ScopeSet} from "../request/ScopeSet";
+import {GrantType} from "../utils/Constants";
+
+/**
+ * OAuth2.0 refresh token client
+ */
+export class RefreshTokenClient extends BaseClient {
```

Comment:
Why are we making a separate client for this? Shouldn't token renewal just be another API on the DeviceCodeClient or AuthCodeClient?

Reply:
I actually think this may be a good idea, but if we are doing this I'd like to see it extended to all flows (including the browser flow). We should also try to take more advantage of subclassing and protected functions.

Reply:
Refresh token it's own Oauth2.0 grant, separate from the others, and therefore it makes sense to have it's own client.

Reply:
Agreed but that's not how we have designed up until now. As I said, it is a good idea since refresh_token grant is pretty much the same across all libs but we should apply it to the SPA lib as well.